### PR TITLE
[14630] Adding configuration data for external locators feature

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR adds `metatraffic_external_unicast_locators`, `default_external_unicast_locators`, and `ignore_non_matching_locators` to the participant configuration structures.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Fast DDS test suite has been run locally. 
- [ ] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

File: include/fastdds/rtps/attributes/EndpointAttributes.h [hunk 1]
```diff
@@ -57,6 +58,12 @@ class EndpointAttributes
     //!GUID used for persistence
     GUID_t persistence_guid;
 
+    //!The collection of external locators to use for communication.
```
Comment (EduPonz) [inline include/fastdds/rtps/attributes/EndpointAttributes.h:61]: Nit:
```suggestion
    //! The collection of external locators to use for communication.
```

File: include/fastdds/rtps/attributes/EndpointAttributes.h [hunk 2]
```diff
@@ -57,6 +58,12 @@ class EndpointAttributes
...
+    fastdds::rtps::ExternalLocators external_unicast_locators;
+
+    //!Whether locators that don't match with the announced locators should be kept.
```
Comment (EduPonz) [inline include/fastdds/rtps/attributes/EndpointAttributes.h:64]: Nit:
```suggestion
    //! Whether locators that don't match with the announced locators should be kept.
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 3]
```diff
@@ -2801,14 +2811,20 @@ class RTPSEndpointQos
     //!Remote locator list
     rtps::LocatorList remote_locator_list;
 
+    //!The collection of external locators to use for communication.
```
Comment (EduPonz) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:2814]:
```suggestion
    //! The collection of external locators to use for communication.
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 4]
```diff
@@ -2801,14 +2811,20 @@ class RTPSEndpointQos
...
+    fastdds::rtps::ExternalLocators external_unicast_locators;
+
+    //!Whether locators that don't match with the announced locators should be kept.
```
Comment (EduPonz) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:2817]:
```suggestion
    //! Whether locators that don't match with the announced locators should be kept.
```

File: include/fastrtps/attributes/PublisherAttributes.h [hunk 5]
```diff
@@ -84,6 +85,12 @@ class PublisherAttributes
     //!Remote locator list
     rtps::LocatorList_t remoteLocatorList;
 
+    //!The collection of external locators to use for communication.
```
Comment (EduPonz) [inline include/fastrtps/attributes/PublisherAttributes.h:88]:
```suggestion
    //! The collection of external locators to use for communication.
```

File: include/fastrtps/attributes/PublisherAttributes.h [hunk 6]
```diff
@@ -84,6 +85,12 @@ class PublisherAttributes
...
+    fastdds::rtps::ExternalLocators external_unicast_locators;
+
+    //!Whether locators that don't match with the announced locators should be kept.
```
Comment (EduPonz) [inline include/fastrtps/attributes/PublisherAttributes.h:91]:
```suggestion
    //! Whether locators that don't match with the announced locators should be kept.
```

File: include/fastrtps/attributes/SubscriberAttributes.h [hunk 7]
```diff
@@ -39,93 +37,120 @@ namespace fastrtps {
...
-
-    private:
-        //!User Defined ID, used for StaticEndpointDiscovery, default value -1.
-        int16_t m_userDefinedID;
-
-        //!Entity ID, if the user want to specify the EntityID of the enpoint, default value -1.
-        int16_t m_entityID;
+public:
+
+    //!Topic Attributes
```
Comment (EduPonz) [inline include/fastrtps/attributes/SubscriberAttributes.h:42]: Everywhere here
```suggestion
    //! Topic Attributes
```

File: include/fastrtps/attributes/SubscriberAttributes.h [hunk 8]
```diff
@@ -39,93 +37,120 @@ namespace fastrtps {
...
+    rtps::LocatorList_t unicastLocatorList;
+
+    //!Multicast locator list
+    rtps::LocatorList_t multicastLocatorList;
+
+    //!Remote locator list
+    rtps::LocatorList_t remoteLocatorList;
+
+    //!The collection of external locators to use for communication.
+    fastdds::rtps::ExternalLocators external_unicast_locators;
```
Comment (EduPonz) [inline include/fastrtps/attributes/SubscriberAttributes.h:61]: Please update the design with this modification

File: include/fastrtps/attributes/PublisherAttributes.h (hunk 5 above)
Comment (EduPonz) [inline include/fastrtps/attributes/PublisherAttributes.h:88]: Please update design with this modification

**Review by EduPonz — CHANGES_REQUESTED** [review]
Nits

**Comment by MiguelCompany** [comment]
@EduPonz This is ready for another review

**Review by EduPonz — APPROVED** [review]

**Pull request merged**